### PR TITLE
feat: define single prop and emit macro

### LINKED ISSUE #294: Define single prop & emits macro `defineProp` & `defineEmit`
### Clear and concise description of the problem

Hi Friend, thanks for the good lib!

So I was thinking that would be nice to have a way to define single `props and emits`  inside the component.

Today we need to declare all props & emits into a single variable and because of this, we will always have a top section in the component where we will define props and emits that will be linked to all code of our components even when following “Logical Concern” guidelines.

### Suggested solution

So would be nice if we could have a macro to declare a single prop or emit for a variable.

Proposal:
```ts
// Props

const count = defineProp(1) // default value 1

const count = defineProp<number>() // prop type number

// Emits

const updateCount = defineEmit('update:count')

const updateCount = defineEmit<number>('update:count') // emit type arg number

const updateCount= defineEmit('update:count', (value: number) => value < 3)) // validation
```

This way we can eliminate the need to have a top section to declare props & emits and we can organize better the code.

## Example:

Before:

```ts
// Props & emits

const props = defineProps(['product', 'category'])

const emit = defineEmits(['update:product', 'update:category'])

// Products

const selectedProduct = computed({
    get: () => props.product,
    set: (v) => emit('update:product', v)
})

// Categories

const selectedCategory= computed({
    get: () => props.category,
    set: (v) => emit('update:category', v)
})

```

After
```ts
// Products

const product = defineProp('product')

const updateProduct = defineEmit('update:product')

const selectedProduct = computed({
    get: () => product.value,
    set: (v) => updateProduct(v)
})

// Categories

const category = defineProp('category')

const updateCategory = defineEmit('update:category')

const selectedCategory= computed({
    get: () => category.value,
    set: (v) => updateCategory(v)
})


```

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by zzhenryquezz**
### Description

This PR will add 2 new macros to the project, `defineProp` & `defineEmit`.

This will allow users to define props & emits multiple times and also organize better the code

### Linked Issues

Fix #294

### Additional context

I use `defineProps` package as a base to create these macros, so please check if is everything ok with file/folder structure and other definitions like packages.json name, tags, etc..,

### CODE REVIEW

File: packages/define-model/tsup.config.bundled_f6zonsim9ra.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// ../../tsup.config.ts
```
Comment (sxzz) [inline packages/define-model/tsup.config.bundled_f6zonsim9ra.mjs:1]: Please remove this file

File: packages/define-emit/package.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (sxzz) [inline packages/define-emit/package.json:1]: I think we can combine these two packages into one package. `@vue-macros/single-define`

Comment (zzhenryquezz) [reply packages/define-emit/package.json:1]: ok, I will make the changes and send till next Saturday

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
Thanks for your awesome work! It has been released as `v1.10.2`, but there are no docs about it yet. Would you like to add some docs?

**Comment by zzhenryquezz** [comment]
> Thanks for your awesome work! It has been released as `v1.10.2`, but there are no docs about it yet. Would you like to add some docs?

Sure I will make it on the weekend

**Pull request merged**